How to implement hint  sharding algorithm?

version:5.3.2
question:How to implement hint  sharding algorithm?
The official website documents are as follows:
```yaml
rules:
- !SHARDING
  tables:
    t_order:
      actualDataNodes: demo_ds_${0..1}.t_order_${0..1}
      databaseStrategy:
        hint:
          algorithmClassName: xxx.xxx.xxx.HintXXXAlgorithm
      tableStrategy:
        hint:
          algorithmClassName: xxx.xxx.xxx.HintXXXAlgorithm
  defaultTableStrategy:
    none:
  defaultKeyGenerateStrategy:
    type: SNOWFLAKE
    column: order_id

props:
    sql-show: true

```
If the configuration is as follows, an error will be displayed:
```
Caused by: org.yaml.snakeyaml.error.YAMLException: Unable to find property 'algorithmClassName' on class: org.apache.shardingsphere.sharding.yaml.config.strategy.sharding.YamlHintShardingStrategyConfiguration
	at org.yaml.snakeyaml.introspector.PropertyUtils.getProperty(PropertyUtils.java:155)
	at org.yaml.snakeyaml.introspector.PropertyUtils.getProperty(PropertyUtils.java:145)
	at org.yaml.snakeyaml.constructor.Constructor$ConstructMapping.getProperty(Constructor.java:337)
	at org.yaml.snakeyaml.constructor.Constructor$ConstructMapping.constructJavaBean2ndStep(Constructor.java:261)
	... 79 more
```

According to error description，check YamlHintShardingStrategyConfiguration properties，change algorithmClassName to shardingAlgorithmName，execute or report an error, as follows:
```
Caused by: org.apache.shardingsphere.sharding.exception.metadata.MissingRequiredShardingAlgorithmException: `com.dragon.shardingsphere.algorithm.hint.UserInfoHintShardingAlgorithm` algorithm does not exist in database `logic_db`.
	at org.apache.shardingsphere.sharding.checker.ShardingRuleConfigurationChecker.lambda$checkShardingStrategy$6(ShardingRuleConfigurationChecker.java:106)
	at org.apache.shardingsphere.infra.util.exception.ShardingSpherePreconditions.checkState(ShardingSpherePreconditions.java:41)
	at org.apache.shardingsphere.sharding.checker.ShardingRuleConfigurationChecker.checkShardingStrategy(ShardingRuleConfigurationChecker.java:105)
	at org.apache.shardingsphere.sharding.checker.ShardingRuleConfigurationChecker.checkTableConfiguration(ShardingRuleConfigurationChecker.java:65)
	at org.apache.shardingsphere.sharding.checker.ShardingRuleConfigurationChecker.check(ShardingRuleConfigurationChecker.java:51)
	at org.apache.shardingsphere.sharding.checker.ShardingRuleConfigurationChecker.check(ShardingRuleConfigurationChecker.java:44)
	at org.apache.shardingsphere.infra.rule.builder.database.DatabaseRulesBuilder.build(DatabaseRulesBuilder.java:62)
	at org.apache.shardingsphere.infra.metadata.database.ShardingSphereDatabase.create(ShardingSphereDatabase.java:87)
	at org.apache.shardingsphere.metadata.factory.ExternalMetaDataFactory.createGenericDatabases(ExternalMetaDataFactory.java:79)
	at org.apache.shardingsphere.metadata.factory.ExternalMetaDataFactory.create(ExternalMetaDataFactory.java:67)
	at org.apache.shardingsphere.mode.metadata.MetaDataContextsFactory.create(MetaDataContextsFactory.java:91)
	at org.apache.shardingsphere.mode.metadata.MetaDataContextsFactory.create(MetaDataContextsFactory.java:68)
	at org.apache.shardingsphere.mode.manager.standalone.StandaloneContextManagerBuilder.build(StandaloneContextManagerBuilder.java:53)
	at org.apache.shardingsphere.driver.jdbc.core.datasource.ShardingSphereDataSource.createContextManager(ShardingSphereDataSource.java:77)
	at org.apache.shardingsphere.driver.jdbc.core.datasource.ShardingSphereDataSource.<init>(ShardingSphereDataSource.java:65)
	at org.apache.shardingsphere.driver.api.ShardingSphereDataSourceFactory.createDataSource(ShardingSphereDataSourceFactory.java:93)
	at org.apache.shardingsphere.driver.api.yaml.YamlShardingSphereDataSourceFactory.createDataSource(YamlShardingSphereDataSourceFactory.java:153)
	at org.apache.shardingsphere.driver.api.yaml.YamlShardingSphereDataSourceFactory.createDataSource(YamlShardingSphereDataSourceFactory.java:95)
	at org.apache.shardingsphere.driver.jdbc.core.driver.DriverDataSourceCache.createDataSource(DriverDataSourceCache.java:51)
	at java.base/java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1708)
	at org.apache.shardingsphere.driver.jdbc.core.driver.DriverDataSourceCache.get(DriverDataSourceCache.java:45)
	at org.apache.shardingsphere.driver.ShardingSphereDriver.connect(ShardingSphereDriver.java:51)
	at com.zaxxer.hikari.util.DriverDataSource.getConnection(DriverDataSource.java:121)
	at com.zaxxer.hikari.pool.PoolBase.newConnection(PoolBase.java:359)
	at com.zaxxer.hikari.pool.PoolBase.newPoolEntry(PoolBase.java:201)
	at com.zaxxer.hikari.pool.HikariPool.createPoolEntry(HikariPool.java:470)
	at com.zaxxer.hikari.pool.HikariPool.checkFailFast(HikariPool.java:561)
	... 30 more
```

--- a/docs/document/content/user-manual/shardingsphere-jdbc/special-api/sharding/hint.cn.md
+++ b/docs/document/content/user-manual/shardingsphere-jdbc/special-api/sharding/hint.cn.md
@@ -23,6 +23,11 @@ Hint 的主要使用场景：
 ### 规则配置
 
 Hint 分片算法需要用户实现 `org.apache.shardingsphere.sharding.api.sharding.hint.HintShardingAlgorithm` 接口。
+`org.apache.shardingsphere.sharding.api.sharding.hint.HintShardingAlgorithm` 存在两个内置实现为，
+
+- `org.apache.shardingsphere.sharding.algorithm.sharding.hint.HintInlineShardingAlgorithm`
+- `org.apache.shardingsphere.sharding.algorithm.sharding.classbased.ClassBasedShardingAlgorithm`
+
 Apache ShardingSphere 在进行路由时，将会从 HintManager 中获取分片值进行路由操作。
 
 参考配置如下：
@@ -35,10 +40,22 @@ rules:
       actualDataNodes: demo_ds_${0..1}.t_order_${0..1}
       databaseStrategy:
         hint:
-          algorithmClassName: xxx.xxx.xxx.HintXXXAlgorithm
+          shardingColumn: order_id
+          shardingAlgorithmName: hint_class_based
       tableStrategy:
         hint:
-          algorithmClassName: xxx.xxx.xxx.HintXXXAlgorithm
+          shardingColumn: order_id
+          shardingAlgorithmName: hint_inline
+  shardingAlgorithms:
+    hint_class_based:
+      type: CLASS_BASED
+      props:
+        strategy: STANDARD
+        algorithmClassName: xxx.xxx.xxx.HintXXXAlgorithm
+    hint_inline:
+      type: HINT_INLINE
+      props:
+        algorithm-expression: t_order_$->{value % 4}
   defaultTableStrategy:
     none:
   defaultKeyGenerateStrategy:

--- a/docs/document/content/user-manual/shardingsphere-jdbc/special-api/sharding/hint.en.md
+++ b/docs/document/content/user-manual/shardingsphere-jdbc/special-api/sharding/hint.en.md
@@ -22,31 +22,48 @@ Main application scenarios for Hint:
 
 ### Hint Configuration
 
-Hint algorithms require users to implement the interface of `org.apache.shardingsphere.api.sharding.hint.HintShardingAlgorithm`. 
+Hint algorithms require users to implement the interface of `org.apache.shardingsphere.api.sharding.hint.HintShardingAlgorithm`.
+`org.apache.shardingsphere.sharding.api.sharding.hint.HintShardingAlgorithm` has two built-in implementations,
+
+- `org.apache.shardingsphere.sharding.algorithm.sharding.hint.HintInlineShardingAlgorithm`
+- `org.apache.shardingsphere.sharding.algorithm.sharding.classbased.ClassBasedShardingAlgorithm`
+
 Apache ShardingSphere will acquire sharding values from HintManager to route.
 
 Take the following configurations for reference:
 
 ```yaml
 rules:
-- !SHARDING
-  tables:
-    t_order:
-      actualDataNodes: demo_ds_${0..1}.t_order_${0..1}
-      databaseStrategy:
-        hint:
+  - !SHARDING
+    tables:
+      t_order:
+        actualDataNodes: demo_ds_${0..1}.t_order_${0..1}
+        databaseStrategy:
+          hint:
+            shardingColumn: order_id
+            shardingAlgorithmName: hint_class_based
+        tableStrategy:
+          hint:
+            shardingColumn: order_id
+            shardingAlgorithmName: hint_inline
+    shardingAlgorithms:
+      hint_class_based:
+        type: CLASS_BASED
+        props:
+          strategy: STANDARD
           algorithmClassName: xxx.xxx.xxx.HintXXXAlgorithm
-      tableStrategy:
-        hint:
-          algorithmClassName: xxx.xxx.xxx.HintXXXAlgorithm
-  defaultTableStrategy:
-    none:
-  defaultKeyGenerateStrategy:
-    type: SNOWFLAKE
-    column: order_id
+      hint_inline:
+        type: HINT_INLINE
+        props:
+          algorithm-expression: t_order_$->{value % 4}
+    defaultTableStrategy:
+      none:
+    defaultKeyGenerateStrategy:
+      type: SNOWFLAKE
+      column: order_id
 
 props:
-    sql-show: true
+  sql-show: true
 ```
 
 ### Get HintManager

